File: remoboard.py
import requests 


class RemoboardClient:
    def __init__(self, url):
        self.url = url
        self.apiurl = url + '/api'
        self.key = '##rkb-1l0v3y0u3000##'

    def ping(self):
        data = {
            'msg': 'input' + self.key + ''
        }
        r = requests.post(self.apiurl, data)
        if r.status_code != 200:
            return False
        return r.text == '{ "result": 0 }'

    def input(self, text):
        data = {
            'msg': 'input' + self.key + text 
        }
        r = requests.post(self.apiurl, data)
        if r.status_code != 200:
            return False
        return r.text == '{ "result": 0 }'

    def delete(self):
        data = {
            'msg': 'input-delete' + self.key + ''
        }
        r = requests.post(self.apiurl, data)
        if r.status_code != 200:
            return False
        return r.text == '{ "result": 0 }'

    def _move(self, direction):
        data = {
            'msg': 'move-' + direction + self.key + '' 
        }
        r = requests.post(self.apiurl, data)
        if r.status_code != 200:
            return False
        return r.text == '{ "result": 0 }'

    def move_up(self):
        return self._move('up')

    def move_down(self):
        return self._move('down')

    def move_left(self):
        return self._move('left')

    def move_right(self):
        return self._move('right')


def main():
    c = RemoboardClient('http://192.168.31.80:7777')

    if not c.ping():
        print('pint failed')
        return

    print('ping ready')
    c.input('hello')
    c.delete()
    c.move_left()
    c.move_right()
    c.move_up()
    c.move_down()

    print('done')
    

if __name__ == '__main__':
    main()

File: nodejs/remoboard.js



var request = require('request');


class RemoboardClient {
    constructor(url) {
        this.url = url;
        this.apiurl = url + '/api';
        this.key = '##rkb-1l0v3y0u3000##';
    }

    ping() {
        request.post(
            this.apiurl,
            {  form: { 
                msg : 'input' + this.key + '' 
            }},
            function (error, response, body) {
                if (!error && response.statusCode == 200) {
                    console.log('ready')
                } else {
                    console.log('error:' + error);
                }
            }
        );
    }

    input(text) {
        request.post(
            this.apiurl,
            {  form: { 
                msg : 'input' + this.key + text
            }},
            function (error, response, body) {
                if (!error && response.statusCode == 200) {
                    console.log('ready')
                } else {
                    console.log('error:' + error);
                }
            }
        );
    }

    delete() {
        request.post(
            this.apiurl,
            {  form: { 
                msg : 'input-delete' + this.key + ''
            }},
            function (error, response, body) {
                if (!error && response.statusCode == 200) {
                    console.log('ready')
                } else {
                    console.log('error:' + error);
                }
            }
        );
    }

    _move(direction) {
        request.post(
            this.apiurl,
            {  form: { 
                msg : 'move-' + direction + this.key + ''
            }},
            function (error, response, body) {
                if (!error && response.statusCode == 200) {
                    console.log('ready')
                } else {
                    console.log('error:' + error);
                }
            }
        );
    }

    move_up() { this._move('up') }
    move_down() { this._move('down') }
    move_left() { this._move('left') }
    move_right() { this._move('right') }

}


c = new RemoboardClient('http://192.168.31.80:7777');
c.ping();
c.input('hello');
c.delete();
c.move_up();
c.move_down();
c.move_left();
c.move_right();


console.log('done');
